consensus/src/experimental/buffer_item.rs

```diff
@@ -137,17 +159,21 @@ impl BufferItem {
                     .check_voting_power(validator)
                     .is_ok()
                 {
-                    Self::Aggregated(Box::new(AggregatedBufferItem {
-                        executed_blocks: signed_item.executed_blocks,
-                        aggregated_proof: signed_item.commit_proof,
-                        callback: signed_item.callback,
-                    }))
+                    (
+                        Self::Aggregated(Box::new(AggregatedBufferItem {
+                            executed_blocks: signed_item.executed_blocks,
+                            aggregated_proof: signed_item.commit_proof,
+                            callback: signed_item.callback,
+                        })),
+                        true,
+                    )
                 } else {
-                    Self::Signed(Box::new(signed_item))
+                    (Self::Signed(Box::new(signed_item)), false)
                 }
             }
             _ => {
-                panic!("Only signed buffer items can advance to aggregated blocks.")
+                // we do not panic here since this is a try function
```

Comment:
only executed may succeed here, ordered should still panic?

Reply:
Good point. Previously I thought it is okay to also try ordered item since it is possible that an ordered item might have enough signatures on some commit info. It is in the TODO at the beginning of this function. Now I think maybe we should at least let the node execute the transactions to make sure the commit info is consistent.

Reply:
We should allow ordered items to gather signatures. So we just ignore them here in trying advancing to aggregated.

Reply:
When advanced to signed item, we will check if the signatures are valid or not.